src/builtins/BuiltinTemporal.cpp

```diff
@@ -0,0 +1,286 @@
+    IteratorRecord* iteratorRecord = IteratorObject::getIterator(state, argv[0]);
+    ValueVector fieldNames;
+    Optional<Object*> next;
+
+    while (true) {
+        next = IteratorObject::iteratorStep(state, iteratorRecord);
+        if (!next.hasValue()) {
+            break;
+        }
+        Value nextValue = IteratorObject::iteratorValue(state, next->asObject());
+        if (!nextValue.isString()) {
+            Value throwCompletion = ErrorObject::createError(state, ErrorObject::TypeError, new ASCIIString("Value is not a string"));
+            return IteratorObject::iteratorClose(state, iteratorRecord, throwCompletion, true);
+        }
+        for (unsigned int i = 0; i < fieldNames.size(); ++i) {
+            if (fieldNames[i].equalsTo(state, nextValue)) {
+                Value throwCompletion = ErrorObject::createError(state, ErrorObject::RangeError, new ASCIIString("Duplicated keys"));
+                return IteratorObject::iteratorClose(state, iteratorRecord, throwCompletion, true);
+            }
+        }
+        if (!(nextValue.asString()->equals("year")
+              || nextValue.asString()->equals("month")
+              || nextValue.asString()->equals("monthCode")
+              || nextValue.asString()->equals("day")
+              || nextValue.asString()->equals("hour")
+              || nextValue.asString()->equals("second")
+              || nextValue.asString()->equals("millisecond")
+              || nextValue.asString()->equals("microsecond")
+              || nextValue.asString()->equals("nanosecond"))) {
+            Value throwCompletion = ErrorObject::createError(state, ErrorObject::RangeError, new ASCIIString("Invalid key"));
+            return IteratorObject::iteratorClose(state, iteratorRecord, throwCompletion, true);
+        }
+        fieldNames.pushBack(nextValue);
+    }
+    return Object::createArrayFromList(state, fieldNames);
+}
+
+static Value builtinTemporalCalendarPrototypeMergeFields(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    ASSERT(thisValue.asObject()->asTemporalCalendarObject()->getIdentifier()->equals("iso8601"));
+    if (argc < 2) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Too few arguments");
+    }
+    Value fields = argv[0];
+    Value additionalFields = argv[1];
+    return TemporalCalendar::defaultMergeFields(state, fields, additionalFields);
+}
+
+static Value builtinTemporalCalendarToString(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return Value(thisValue.asObject()->asTemporalCalendarObject()->getIdentifier());
+}
+
+static Value builtinTemporalCalendarToJSON(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return Value(thisValue.asObject()->asTemporalCalendarObject()->getIdentifier());
```

Comment:
builtinTemporalCalendarToString / builtinTemporalCalendarToJSON are the same. Is this intentional?

Reply:
Yes this is intentional.